ledger/pulsemanager/pulsemanager.go

```diff
@@ -116,20 +117,40 @@ func (m *PulseManager) createDrop(ctx context.Context, lastSlotPulse *storage.Pu
 	jetID := core.TODOJetID
 	prevDrop, err := m.db.GetDrop(ctx, jetID, *lastSlotPulse.Prev)
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, errors.Wrap(err, "[ createDrop ] Can't GetDrop")
 	}
-	drop, messages, err = m.db.CreateDrop(ctx, jetID, lastSlotPulse.Pulse.PulseNumber, prevDrop.Hash)
+	drop, messages, dropSize, err := m.db.CreateDrop(ctx, jetID, lastSlotPulse.Pulse.PulseNumber, prevDrop.Hash)
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, errors.Wrap(err, "[ createDrop ] Can't CreateDrop")
 	}
 	err = m.db.SetDrop(ctx, jetID, drop)
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, errors.Wrap(err, "[ createDrop ] Can't SetDrop")
 	}
 
 	dropSerialized, err = jet.Encode(drop)
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, errors.Wrap(err, "[ createDrop ] Can't Encode")
+	}
+
+	dropSizeData := jet.DropSizeData{
+		JetID:    jetID,
+		PulseNo:  lastSlotPulse.Pulse.PulseNumber, // TODO: is pulse correct ?
+		DropSize: dropSize,
+	}
+	signature, err := m.CryptographyService.Sign(dropSizeData.Bytes(ctx))
```

Comment:
I think it's a bad practice to sign serialized struct which in itself contains signature field. It's probably better to calculate hash from each field and sign it.

Reply:
Also, this way we can get rid of DropSizeData struct.